<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kathi">

<title>4d-Report</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="quarto.tillmann_files/libs/clipboard/clipboard.min.js"></script>
<script src="quarto.tillmann_files/libs/quarto-html/quarto.js"></script>
<script src="quarto.tillmann_files/libs/quarto-html/popper.min.js"></script>
<script src="quarto.tillmann_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="quarto.tillmann_files/libs/quarto-html/anchor.min.js"></script>
<link href="quarto.tillmann_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="quarto.tillmann_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="quarto.tillmann_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="quarto.tillmann_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="quarto.tillmann_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">4d-Report</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Kathi </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="environment" class="level1">
<h1>1. Environment</h1>
<section id="a-load-libraries" class="level2">
<h2 class="anchored" data-anchor-id="a-load-libraries">1.a) Load Libraries</h2>
<p>#add Seurat 5 to your working environment</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">.libPaths</span>(<span class="fu">c</span>(<span class="st">"/lisc/user/tillmann"</span>,<span class="st">"/home/apps/seurat5/5.0.1-4.3.2"</span>)) </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#* set working directory: </span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">'/scratch/course/2023w300106/tillmann/'</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#change language</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.setenv</span>(<span class="at">LANG =</span> <span class="st">"en"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(easypackages)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">libraries</span>(<span class="st">"readxl"</span>,<span class="st">"RColorBrewer"</span>,<span class="st">'patchwork'</span>,<span class="st">'dplyr'</span>,<span class="st">'viridis'</span>,<span class="st">'ggplot2'</span>,<span class="st">'pals'</span>,<span class="st">'clustree'</span>,<span class="st">'ape'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: readxl</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: RColorBrewer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: patchwork</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: dplyr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridisLite</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggplot2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: pals</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'pals'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridis':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridisLite':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: clustree</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggraph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ape</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ape'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    where</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>All packages loaded successfully</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"Seurat"</span>, <span class="at">lib.loc =</span> <span class="st">"/lisc/app/seurat5/5.0.1-4.3.2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SeuratObject</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: sp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'sp'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggraph':

    geometry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'SeuratObject' was built with package 'Matrix' 1.6.4 but the current
version is 1.6.5; it is recomended that you reinstall 'SeuratObject' as
the ABI for 'Matrix' may have changed</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SeuratObject'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    intersect</code></pre>
</div>
</div>
</section>
<section id="b-colour-palettes--" class="level2">
<h2 class="anchored" data-anchor-id="b-colour-palettes--">1.b) Colour Palettes â€”-</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 'rainbow' gene color palette: </span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>gene.cp <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'lightgrey'</span>, <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">11</span> , <span class="st">"Spectral"</span>))) </span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># can print any colour palette: </span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">pal.bands</span>(gene.cp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># maximally unique colour palette for clustering: </span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>clust.cp.separate <span class="ot">=</span> <span class="fu">unique</span> (<span class="fu">c</span>(<span class="fu">cols25</span>(<span class="dv">25</span>), <span class="fu">glasbey</span>(<span class="dv">32</span>), <span class="fu">alphabet</span>(<span class="dv">26</span>)))</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pal.bands</span>(clust.cp.separate,<span class="fu">cols25</span>(<span class="dv">25</span>), <span class="fu">glasbey</span>(<span class="dv">32</span>), <span class="fu">alphabet</span>(<span class="dv">26</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-2-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># gradient colour palette for clustering: </span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>clust.cp.graded <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">c</span>(<span class="fu">stepped3</span>(<span class="dv">20</span>), <span class="fu">stepped2</span>(<span class="dv">20</span>), <span class="fu">stepped</span>(<span class="dv">20</span>)))</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pal.bands</span>(clust.cp.graded,<span class="fu">stepped3</span>(<span class="dv">20</span>), <span class="fu">stepped2</span>(<span class="dv">20</span>), <span class="fu">stepped</span>(<span class="dv">20</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-2-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set a colour palet for libraries (relevant for merged objects: week 2) </span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>LibCP <span class="ot">=</span> <span class="fu">brewer.paired</span>(<span class="dv">12</span>) </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pal.bands</span>(LibCP,<span class="fu">brewer.paired</span>(<span class="dv">20</span>),<span class="fu">brewer.paired</span>(<span class="dv">40</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-2-4.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="c-gene-lists" class="level2">
<h2 class="anchored" data-anchor-id="c-gene-lists">1.c) Gene Lists</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a> annotations <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"Nv2.annotations.xlsx"</span>) <span class="co">#This throws some warnings</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting numeric in J24541 / R24541C10: got '-'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting numeric in J24542 / R24542C10: got '-'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting numeric in J24543 / R24543C10: got '-'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting numeric in J24544 / R24544C10: got '-'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting numeric in J24545 / R24545C10: got '-'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting numeric in J24546 / R24546C10: got '-'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting numeric in J24547 / R24547C10: got '-'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting numeric in J24548 / R24548C10: got '-'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting numeric in J24549 / R24549C10: got '-'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
* `NR_Top_Blastx_Desc` -&gt; `NR_Top_Blastx_Desc...9`
* `NR_Top_Blastx_Desc` -&gt; `NR_Top_Blastx_Desc...16`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"Nv2.annotations.xlsx"</span>, <span class="at">sheet =</span> <span class="st">'cellranger'</span>) <span class="co"># this is output from the cellranger directory</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  NVEs <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"Nv2.annotations.xlsx"</span>, <span class="at">sheet =</span> <span class="st">'NvNames_NVE'</span>) <span class="co"># this is a list of confirmed gene names from a previous gene models</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="do">### lesson: indexing ----</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pull out coordinates for your table: which line corresponds to model with a confirmed gene name</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  ind <span class="ot">=</span> <span class="fu">match</span>(NVEs<span class="sc">$</span>NVE, annotations<span class="sc">$</span>NVE, <span class="at">nomatch =</span> <span class="dv">0</span>)</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># loop through each gene model and update with the new gene name</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ind))</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># go to row XX in the annotations and replace the name from the NVE list:</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>    annotations<span class="sc">$</span>gene_short_name[ind[i]] <span class="ot">=</span> NVEs<span class="sc">$</span>gene_short_name[i]</span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a>  Nv2 <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">"Nv2.annotations.xlsx"</span>, <span class="at">sheet =</span> <span class="st">'names_Nv2'</span>) <span class="co"># this is a list of confirmed Nv2 model identities</span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a>  <span class="co">#* **Alternative** incorporate the index directly into the command:</span></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a>  annotations<span class="sc">$</span>gene_short_name[<span class="fu">match</span>(Nv2<span class="sc">$</span>Nv2, annotations<span class="sc">$</span>geneID)] <span class="ot">=</span> Nv2<span class="sc">$</span>gene_short_name</span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># row names (genes) must be unique for Seurat:</span></span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a>  annotations<span class="sc">$</span>gene_short_name <span class="ot">=</span> <span class="fu">make.unique</span>(annotations<span class="sc">$</span>gene_short_name)</span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># generate a new gene name and annotation file corresponding to the matrix:</span></span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">=</span> <span class="fu">merge</span>(genes, <span class="co">#cellranger list</span></span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a>                annotations, <span class="co">#gene information table</span></span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true" tabindex="-1"></a>                <span class="at">by =</span> <span class="st">"geneID"</span>, <span class="co">#this is the common denominator!</span></span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true" tabindex="-1"></a>                <span class="at">all.x =</span> T, <span class="co">#don't skip anything from cell ranger</span></span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true" tabindex="-1"></a>                <span class="at">sort =</span> F) <span class="co"># **DO NOT** reorder the genes file!!!</span></span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># keep only interesting / useful columns here</span></span>
<span id="cb45-31"><a href="#cb45-31" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">=</span> genes[, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)] </span>
<span id="cb45-32"><a href="#cb45-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-33"><a href="#cb45-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># generate a list of genes with DNAbinding domains:</span></span>
<span id="cb45-34"><a href="#cb45-34" aria-hidden="true" tabindex="-1"></a>  TF_list <span class="ot">=</span> annotations<span class="sc">$</span>gene_short_name[annotations<span class="sc">$</span>DNABinding <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb45-35"><a href="#cb45-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-36"><a href="#cb45-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-37"><a href="#cb45-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># clean up the workspace</span></span>
<span id="cb45-38"><a href="#cb45-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(i, ind, NVEs, Nv2)</span>
<span id="cb45-39"><a href="#cb45-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-40"><a href="#cb45-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save these for recall / use later</span></span>
<span id="cb45-41"><a href="#cb45-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">save</span>(<span class="at">list =</span> <span class="fu">c</span>(<span class="st">'annotations'</span>, <span class="st">'genes'</span>, <span class="st">'TF_list'</span>),<span class="at">file =</span> <span class="st">'results/Genes.Nv2.RData'</span>)</span>
<span id="cb45-42"><a href="#cb45-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-43"><a href="#cb45-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">load</span> (<span class="st">'results/Genes.Nv2.RData'</span>) <span class="co">#otherwise just load these!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="load-raw-data-and-create-seurat-object" class="level1">
<h1>2. Load raw data and create Seurat Object</h1>
<section id="a-load-raw-data-and-check-if-genes-file-and-data-matrix-are-identical" class="level2">
<h2 class="anchored" data-anchor-id="a-load-raw-data-and-check-if-genes-file-and-data-matrix-are-identical">2.a) Load raw data and check if genes file and data matrix are identical</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>raw.data1 <span class="ot">&lt;-</span> <span class="fu">Read10X</span>(<span class="at">data.dir =</span><span class="st">'results/4d/outs/filtered_feature_bc_matrix/'</span>)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw.data1)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NV2.1" "NV2.2" "NV2.3" "NV2.4" "NV2.5"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(genes<span class="sc">$</span>geneID, <span class="fu">rownames</span>(raw.data1))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw.data1) <span class="ot">&lt;-</span> genes<span class="sc">$</span>gene_short_name</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="b-remove-gene-rows-with-no-mapping-data-in-data-matrix-and-genes-file" class="level2">
<h2 class="anchored" data-anchor-id="b-remove-gene-rows-with-no-mapping-data-in-data-matrix-and-genes-file">2.b) Remove gene rows with no mapping data in data matrix and genes file</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>  keep <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">rowSums</span>(raw.data1) <span class="sc">&gt;=</span> <span class="dv">3</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">length</span>(keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 20387</code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>  raw.data1 <span class="ot">=</span> raw.data1[keep, ]</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dim</span> (raw.data1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 20387 10000</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> genes[keep, ]</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dim</span> (genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 20387     8</code></pre>
</div>
</div>
</section>
<section id="c-generate-seurat-object" class="level2">
<h2 class="anchored" data-anchor-id="c-generate-seurat-object">2.c) Generate Seurat object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>data1  <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> raw.data1, <span class="at">project =</span> <span class="st">"4d"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="d-look-at-the-raw-data-distribution" class="level2">
<h2 class="anchored" data-anchor-id="d-look-at-the-raw-data-distribution">2.d) Look at the raw data distribution</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FeatureScatter</span>(data1, <span class="at">feature1 =</span> <span class="st">"nCount_RNA"</span>, <span class="at">feature2 =</span> <span class="st">"nFeature_RNA"</span>) <span class="sc">+</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="dv">500</span>, <span class="dv">40000</span>)) <span class="sc">+</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">c</span>(<span class="dv">300</span>, <span class="dv">6000</span>)) <span class="sc">+</span></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stat_smooth</span>(<span class="at">method =</span> lm) <span class="sc">+</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_y_log10</span>()</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="ecut-and-plot" class="level2">
<h2 class="anchored" data-anchor-id="ecut-and-plot">2.e)Cut and Plot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span>   <span class="fu">subset</span>(<span class="at">x =</span> data1, <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">150</span> <span class="sc">&amp;</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>                      nCount_RNA <span class="sc">&lt;</span> <span class="dv">40000</span>) <span class="co">#update for each sample!!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>vl <span class="ot">=</span> <span class="fu">VlnPlot</span>(data1, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">'nFeature_RNA'</span>, <span class="st">'nCount_RNA'</span>),<span class="at">log=</span>F)<span class="co">#Visual guide</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Default search for "data" layer in "RNA" assay yielded no results;
utilizing "counts" layer instead.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># to select the lower limit - cell density</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">=</span> data1<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> nFeature_RNA)) <span class="sc">+</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Cell density"</span>) <span class="sc">+</span></span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="dv">350</span>, <span class="dv">4000</span>)) <span class="co">#can adjust this</span></span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">=</span> data1<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span></span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> nCount_RNA)) <span class="sc">+</span></span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb63-14"><a href="#cb63-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Cell density"</span>) <span class="sc">+</span></span>
<span id="cb63-15"><a href="#cb63-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="dv">750</span>, <span class="dv">40000</span>)) <span class="co">#can adjust this</span></span>
<span id="cb63-16"><a href="#cb63-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(vl <span class="sc">+</span> p1 <span class="sc">+</span> p2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="data-analysis" class="level1">
<h1>3. Data Analysis</h1>
<section id="a-normalize-data" class="level3">
<h3 class="anchored" data-anchor-id="a-normalize-data">3.1a Normalize Data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(data1, <span class="at">scale.factor =</span> <span class="dv">10000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Normalizing layer: counts</code></pre>
</div>
</div>
</section>
<section id="b-find-variable-genes" class="level3">
<h3 class="anchored" data-anchor-id="b-find-variable-genes">3.1b Find variable genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(data1, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Finding variable features for layer counts</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VariableFeaturePlot</span>(data1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
</div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="c-scale" class="level3">
<h3 class="anchored" data-anchor-id="c-scale">3.1c Scale</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(data1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Centering and scaling data matrix</code></pre>
</div>
</div>
</section>
<section id="a-dimension-reduction-pca" class="level3">
<h3 class="anchored" data-anchor-id="a-dimension-reduction-pca">3.2a Dimension Reduction: PCA</h3>
<div class="cell" data-fig.dim="[6,12]">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(data1, <span class="at">pcs.compute =</span> <span class="dv">50</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  ACTC-like-6, NV2.6264, Klf5, ELEL-like-2, MYSA-like-1, CO1A2-like-2, MCPI-like-1, NV2.3734, RIR2-like-1, CSRP1-like-1 
       APLP-like-1, NV2.101, Tpm9, NV2.16006, COA-like-2, COA-like-1, NV2.5844, CO4A2-like-1, CATL-like-3, CKS1-like-4 
       RIR2-like-2, CALUB-like-4, SPRC-like-10, ENKUR-like-1, HEBP2-like-1, VKT2-like-1, TIL-like-1, Hes4, Six3-6, TICN2-like-1 
Negative:  DKK3-like-2, NV2.12690, NV2.24893, Ncol1, DKK3-like-11, LOXA-like-1, NV2.9738, GGT1-like-26, DCBD2-like-1, QRIC2-like-2 
       NV2.9649, SAL-like-1, NV2.17352, GAPR1-like-17, NV2.9741, FBP1-like-25, NV2.9497, ECE2-like-6, NV2.12542, NV2.21582 
       CRA1B-like-2, NCKX2-like-3, NV2.17003, NV2.16646, NV2.9872, NV2.10332, FBP1-like-45, NV2.18954, COHA1-like-2, MATN1-like-2 
PC_ 2 
Positive:  ANO8-like-1, BIR7B-like-1, MCTP1-like-1, NV2.25609, NV2.24075, SEGN-like-4, CALM3-like-6, NV2.160, KCNK9-like-8, TCB1-like-1 
       NV2.18301, R1A-like-2, NV2.25610, FOS-like-1, NV2.9160, NV2.14822, MUC19-like-2, NV2.13572, NV2.23951, MBOA2-like-1 
       Y1101-like-1, NV2.24640, SRX-like-2, LAMA2-like-2.1, T214B-like-1, NV2.11749, NV2.8832, XYNB-like-6.2, TBA-like-2, CEMIP-like-42 
Negative:  TXD12-like-2, NV2.11825, NV2.3555, CALUA-like-5, NV2.10282, PAX2A-like-1, GAPR1-like-1, NV2.6073, NV2.17391, TYRO-like-1 
       FBP1-like-10, VGLU3-like-1, FKBP2-like-1, NOTCH-like-1, P4HTM-like-3, GAPR1-like-57, NV2.13500, NV2.12784, FBP1-like-15, NEP4 
       GAPR1-like-50, PPIB-like-5, TBB2-like-1, ARSB-like-1, NV2.17017, YCXC-like-1, MATN2-like-9, HMCN1-like-34, MLP-like-2, VWDE-like-12 
PC_ 3 
Positive:  NV2.12553, NV2.5190, JUN-like-11, PDIA4-like-2, NV2.9507, NV2.17401, NV2.17398, NV2.12691, NV2.17400, NV2.11394 
       PGCA-like-29, NV2.11725, NV2.2533, VGLU2-like-17, NV2.11718, NV2.5192, Y683-like-1, TXND5-like-1, NV2.17397, CBPD-like-1 
       GAPR1-like-46, KRBBA-like-4, SML-like-8, VGLU1-like-5, CEL3B-like-2, NV2.9513, NV2.12323, CHST1-like-2, HYAL5-like-1, CBPD-like-2 
Negative:  TXD12-like-2, P4HTM-like-3, MSRE-like-3, XYLT-like-2, NV2.10282, NV2.23423, ARSB-like-1, NV2.13500, VWDE-like-12, KCNK9-like-3 
       NV2.6295, GILT-like-1, NV2.12896, NV2.3555, B3GA3-like-1, PAX2A-like-1, CR3L1-like-1, CALUA-like-5, GGT1-like-8, P4HTM-like-16 
       NV2.11825, GAPR1-like-1, COT2-like-1, P4HA2-like-1, P4HTM-like-1, TMEDA-like-1, SL9C1-like-1, G3ST1-like-6, CADN-like-7, NV2.920 
PC_ 4 
Positive:  NV2.22820, Mucin, SPEE-like-2, AHNK-like-2, CRAM-like-6, NV2.21610, ALK-like-1, AHNK-like-9, NV2.17583, NV2.3425 
       NV2.4247, DOX2-like-2, FKB14-like-1, GUNA-like-1, SEGN-like-1, NV2.16639, NV2.7431, GUNA-like-9, B4GT1-like-1, CDCP2-like-10 
       OGFR-like-6, NV2.5209, NV2.19510, HMCN1-like-100, B4GN4-like-25, NEC1-like-2, LAF3-like-3, ACHA7-like-23, NV2.8632, ESRP2-like-1 
Negative:  CO1A2-like-2, COA-like-2, CO4A2-like-1, NV2.16006, COA-like-1, MCPI-like-1, APLP-like-1, NV2.3734, NV2.15882, CSRP3-like-1 
       SPRC-like-10, TICN2-like-1, DAPK3-like-1, CALUB-like-4, Tpm9, CO2A1-like-1, LAMC1-like-1, Perl, CALM-like-38, CSRP1-like-1 
       MCPI-like-2, RHCL2-like-1, COBA1-like-1, CAF1-like-2, P3H2-like-1, HEBP2-like-1, MYSA-like-1, GELS1-like-6, SAP-like-9, BCCT1-like-9 
PC_ 5 
Positive:  NV2.22820, SPEE-like-2, Mucin, NV2.5209, CRAM-like-6, NV2.21610, NV2.17583, AHNK-like-2, ESRP2-like-1, GUNA-like-1 
       AHNK-like-9, NV2.16639, ALK-like-1, DOX2-like-2, FKB14-like-1, NV2.4247, B4GT1-like-1, GUNA-like-9, NV2.3425, NV2.19510 
       SEGN-like-1, MUP4-like-1, OGFR-like-6, CDCP2-like-10, HMCN1-like-100, NV2.7431, STL1-like-2, creb-like2, B4GN4-like-25, ACHA7-like-23 
Negative:  ELEL-like-2, NV2.14822, LAMA2-like-2.1, NV2.18301, NV2.25609, PTPRF-like-27, MUC19-like-2, SEGN-like-4, NV2.25610, NV2.11749 
       AVIL-like-1, MEC10-like-1, SVEP1-like-29, CEMIP-like-42, CF015-like-1, FBP3-like-9, NV2.2136, NPTX2-like-15, NV2.5844, NV2.22235 
       FOXD1-like-1, ISP2-like-2, TRAF4-like-12, BIR7B-like-1, XYNB-like-6.2, OAS1-like-4, ESPB1-like-1, Rx, NV2.160, ANO8-like-1 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">PCAPlot</span>(data1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="576"></p>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">PCHeatmap</span>(data1, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>, <span class="at">nfeatures =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-14-2.png" class="img-fluid" width="576"></p>
</div>
</div>
<p>Plot that again after Clustering with different cluster colors</p>
</section>
<section id="b-select-informative-dimensions" class="level3">
<h3 class="anchored" data-anchor-id="b-select-informative-dimensions">3.2b Select informative dimensions</h3>
<p>##JackStraw - Test:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">JackStraw</span>(data1, <span class="at">dims =</span> <span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">ScoreJackStraw</span>(data1, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">JackStrawPlot</span>(data1, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>, <span class="at">cols =</span> clust.cp.separate)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 42303 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="determine-d" class="level2">
<h2 class="anchored" data-anchor-id="determine-d">Determine d:</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>d <span class="ot">=</span> <span class="fu">as.integer</span>(<span class="fu">which</span>(data1<span class="sc">@</span>reductions<span class="sc">$</span>pca<span class="sc">@</span>stdev <span class="sc">&gt;</span> <span class="dv">2</span>))</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>d</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25</code></pre>
</div>
</div>
</section>
<section id="run-umap" class="level2">
<h2 class="anchored" data-anchor-id="run-umap">Run UMAP</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>    data1,</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">dims =</span> d,</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction =</span> <span class="st">'pca'</span>,</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># reduction.name = 'umap1',</span></span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction.key =</span> <span class="st">'umap1'</span>,</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>    <span class="co">#this is default; can change name to save different maps</span></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>    <span class="co">#the following parameters control your output.</span></span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">seed.use =</span> <span class="dv">7</span>,</span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>    <span class="co">#random starting point; without your maps will differ every time!</span></span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">n.neighbors =</span> 10L,</span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a>    <span class="co">#how many similar cells do you expect</span></span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">spread =</span> <span class="fl">0.2</span>,</span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>    <span class="co">#how big is the axis space: 0.2 for lineages; 1 for separation</span></span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">min.dist =</span> <span class="fl">0.2</span>,</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>    <span class="co">#how close to plot each cell higher=spread</span></span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">local.connectivity =</span> <span class="dv">10</span></span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>  )<span class="co">#overall how connected is the graph</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'
This message will be shown once per session</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>13:16:56 UMAP embedding parameters a = 57.22 b = 1.929</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>13:16:56 Read 5118 rows and found 25 numeric columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>13:16:56 Using Annoy for neighbor search, n_neighbors = 10</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>13:16:56 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>**************************************************|
13:16:57 Writing NN index file to temp file /tmp/Rtmpe1hSKK/fileee8ff1a32d500
13:16:57 Searching Annoy index using 1 thread, search_k = 1000
13:16:58 Annoy recall = 100%
13:16:58 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 10
13:16:58 5118 smooth knn distance failures
13:16:59 Initializing from normalized Laplacian + noise (using RSpectra)
13:16:59 Commencing optimization for 500 epochs, with 75356 positive edges
13:17:09 Optimization finished</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Keys should be one or more alphanumeric characters followed by an
underscore, setting key from umap1 to umap1_</code></pre>
</div>
</div>
</section>
<section id="run-tsne" class="level2">
<h2 class="anchored" data-anchor-id="run-tsne">Run tSNE</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">RunTSNE</span>(</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>    data1,</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">seed.use =</span> <span class="dv">42</span>,</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>    <span class="co">#necessary for reproducibility; try turning it off!</span></span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">perplexity =</span> <span class="dv">100</span>,</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a>    <span class="co">#default 30; higher maintains greater local</span></span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">dims.use =</span> d</span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="print-umap-and-tsne" class="level2">
<h2 class="anchored" data-anchor-id="print-umap-and-tsne">Print UMAp and tSNE</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a> <span class="fu">print</span>(</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FeaturePlot</span>(</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>      data1,</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>      <span class="st">'nFeature_RNA'</span>,</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">reduction =</span> <span class="st">'umap'</span>,</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">'lightgrey'</span>, <span class="st">'darkred'</span>)</span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> <span class="fu">NoAxes</span>() <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span></span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">'nFeatures'</span>, <span class="at">subtitle =</span> <span class="st">'umap'</span>) <span class="sc">&amp;</span></span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a>      <span class="fu">FeaturePlot</span>(</span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a>        data1,</span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">'nFeature_RNA'</span>,</span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">'tsne'</span>,</span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">'lightgrey'</span>, <span class="st">'darkred'</span>)</span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">+</span> <span class="fu">NoAxes</span>() <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span></span>
<span id="cb93-15"><a href="#cb93-15" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">'nFeatures'</span>, <span class="at">subtitle =</span> <span class="st">'tsne'</span>)</span>
<span id="cb93-16"><a href="#cb93-16" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="clustering" class="level3">
<h3 class="anchored" data-anchor-id="clustering">3.3 Clustering</h3>
<section id="a-neighbors" class="level4">
<h4 class="anchored" data-anchor-id="a-neighbors">3.3a Neighbors</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> data1,</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction =</span> <span class="st">"pca"</span>,</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">dims =</span> d,<span class="at">annoy.metric =</span> <span class="st">'cosine'</span>,</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>    <span class="co">#default = 'euclidean'</span></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># see https://www.nature.com/articles/s41592-019-0372-4</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">k.param =</span> <span class="dv">10</span></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing nearest neighbor graph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing SNN</code></pre>
</div>
</div>
</section>
<section id="b-clusters" class="level4">
<h4 class="anchored" data-anchor-id="b-clusters">3.3b Clusters</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>check.resolution <span class="ot">=</span> T <span class="co"># a short script to test and plot different resolutions:</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (check.resolution)</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>    <span class="co">#here you can calculate different clustering resolutions:</span></span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="fl">1.5</span>, <span class="dv">2</span>) <span class="co">#list of resolutions 0.1 to 2 in 0.5 increments</span></span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(res))</span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>      data1 <span class="ot">&lt;-</span></span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>        <span class="fu">FindClusters</span>(</span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>          <span class="at">object =</span> data1,</span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">resolution =</span> res[i],</span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">random.seed =</span> <span class="dv">7</span></span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>      <span class="co">#save all the plots:</span></span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>      p[[i]] <span class="ot">=</span> <span class="fu">DimPlot</span>(</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a>        data1,</span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> T,</span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">5</span>,</span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T,</span>
<span id="cb97-22"><a href="#cb97-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> clust.cp.separate</span>
<span id="cb97-23"><a href="#cb97-23" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span> <span class="fu">NoAxes</span>() <span class="sc">+</span></span>
<span id="cb97-24"><a href="#cb97-24" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">c</span>(res[i], <span class="st">'Clusters | ID'</span>))</span>
<span id="cb97-25"><a href="#cb97-25" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb97-26"><a href="#cb97-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-27"><a href="#cb97-27" aria-hidden="true" tabindex="-1"></a>    <span class="co">#can use clustree to image clustering stability:</span></span>
<span id="cb97-28"><a href="#cb97-28" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> clustree<span class="sc">::</span><span class="fu">clustree</span>(data1, <span class="at">prefix =</span> <span class="st">"RNA_snn_res."</span>)</span>
<span id="cb97-29"><a href="#cb97-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(x)</span>
<span id="cb97-30"><a href="#cb97-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-31"><a href="#cb97-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#choose your desired resolution:</span></span>
<span id="cb97-32"><a href="#cb97-32" aria-hidden="true" tabindex="-1"></a>    p[[<span class="dv">1</span>]] <span class="sc">+</span> p[[<span class="dv">2</span>]] <span class="sc">+</span> p[[<span class="dv">3</span>]] <span class="sc">+</span> p[[<span class="dv">4</span>]]</span>
<span id="cb97-33"><a href="#cb97-33" aria-hidden="true" tabindex="-1"></a>  }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5118
Number of edges: 89761

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9513
Number of communities: 10
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5118
Number of edges: 89761

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8911
Number of communities: 17
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5118
Number of edges: 89761

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8499
Number of communities: 26
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5118
Number of edges: 89761

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8209
Number of communities: 32
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5118
Number of edges: 89761

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7959
Number of communities: 37
Elapsed time: 0 seconds</code></pre>
</div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-23-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="recalculate-from-selected-resolution-01" class="level4">
<h4 class="anchored" data-anchor-id="recalculate-from-selected-resolution-01">Recalculate from selected resolution: 0,1</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span><span class="fu">FindClusters</span>(<span class="at">object =</span> data1,</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">resolution =</span> <span class="fl">0.1</span>,</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">random.seed =</span> <span class="dv">7</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5118
Number of edges: 89761

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9513
Number of communities: 10
Elapsed time: 0 seconds</code></pre>
</div>
</div>
</section>
<section id="relationship-between-clusters" class="level4">
<h4 class="anchored" data-anchor-id="relationship-between-clusters">Relationship between clusters:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">BuildClusterTree</span>(</span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> data1,</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">reorder =</span> <span class="cn">TRUE</span>,</span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">dims =</span> d,</span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">reorder.numeric =</span> T</span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Reordering identity classes and rebuilding tree</code></pre>
</div>
</div>
</section>
<section id="plot" class="level4">
<h4 class="anchored" data-anchor-id="plot">Plot:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>ape<span class="sc">::</span><span class="fu">plot.phylo</span>(data1<span class="sc">@</span>tools<span class="sc">$</span>BuildClusterTree, <span class="at">direction =</span> <span class="st">'right'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="c-cellplots" class="level4">
<h4 class="anchored" data-anchor-id="c-cellplots">3.3c CellPlots</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>    data1,</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> T,</span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.size =</span> <span class="dv">5</span>,</span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">repel =</span> T,</span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> clust.cp.separate</span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">&amp;</span></span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">NoLegend</span>() <span class="sc">&amp;</span> <span class="fu">NoAxes</span>() <span class="sc">&amp;</span></span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">DimPlot</span>(</span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a>      data1,</span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">label =</span> T,</span>
<span id="cb104-12"><a href="#cb104-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">label.size =</span> <span class="dv">5</span>,</span>
<span id="cb104-13"><a href="#cb104-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">repel =</span> T,</span>
<span id="cb104-14"><a href="#cb104-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">cols =</span> clust.cp.separate,</span>
<span id="cb104-15"><a href="#cb104-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">reduction =</span> <span class="st">'umap'</span></span>
<span id="cb104-16"><a href="#cb104-16" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">&amp;</span></span>
<span id="cb104-17"><a href="#cb104-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">NoLegend</span>() <span class="sc">&amp;</span> <span class="fu">NoAxes</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="differential-gene-expression" class="level4">
<h4 class="anchored" data-anchor-id="differential-gene-expression">4.4 Differential gene expression</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a> data1<span class="sc">@</span>active.assay <span class="ot">=</span> <span class="st">'RNA'</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>    all.markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>      data1,</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">logfc.threshold =</span> <span class="dv">1</span>,</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">features =</span> <span class="fu">VariableFeatures</span>(data1),</span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>      <span class="co">#this is faster but restricted to variable gene set.</span></span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">return.thresh =</span> <span class="fl">0.001</span>,</span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">min.pct =</span> <span class="fl">0.2</span>,</span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">only.pos =</span> <span class="cn">TRUE</span>,</span>
<span id="cb105-10"><a href="#cb105-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">max.cells.per.ident =</span> <span class="dv">200</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 5</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 6</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 7</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 8</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 9</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 10</code></pre>
</div>
</div>
</section>
<section id="make-a-gene-list" class="level4">
<h4 class="anchored" data-anchor-id="make-a-gene-list">Make a gene list</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>list <span class="ot">=</span> <span class="cn">NULL</span> </span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">levels</span>(data1<span class="sc">@</span>active.ident)))</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>      {x <span class="ot">=</span> all.markers[<span class="fu">as.numeric</span>(all.markers<span class="sc">$</span>cluster) <span class="sc">==</span> i, ][<span class="dv">1</span><span class="sc">:</span><span class="fu">min</span>(<span class="dv">5</span>, <span class="fu">length</span>(<span class="fu">which</span>(<span class="fu">as.numeric</span>(all.markers<span class="sc">$</span>cluster) <span class="sc">==</span> i))), <span class="dv">7</span>]</span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>list <span class="ot">=</span> <span class="fu">c</span>(list, x)}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-top-5-degs" class="level4">
<h4 class="anchored" data-anchor-id="plot-top-5-degs">Plot Top 5 DEGs</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>DEG.plot <span class="ot">=</span> <span class="fu">DotPlot</span>(</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>      data1,</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">features =</span> <span class="fu">unique</span>(<span class="fu">c</span>(list)),</span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">scale.by =</span> <span class="st">'size'</span> ,</span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">col.min =</span> <span class="dv">0</span>,</span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">col.max =</span> <span class="dv">3</span>,</span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">'lightgrey'</span>, <span class="st">'darkred'</span>)</span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>      <span class="fu">RotatedAxis</span>() <span class="sc">+</span> <span class="fu">FontSize</span>(<span class="dv">6</span>, <span class="dv">6</span>) <span class="sc">+</span> <span class="co">#coord_flip()+</span></span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a>      <span class="co"># NoLegend()+</span></span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">'Top 5 DEGs'</span>, <span class="at">subtitle =</span> <span class="st">'p-val &lt; 0.001'</span>)</span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a>DEG.plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="pharyngeal.epithelia.2-looks-like-it-resembles-a-rather-unspecific-cluster-which-could-be-a-hint-for-bad-filtering.-cluster-doesnt-look-over--or-underclustered." class="level5">
<h5 class="anchored" data-anchor-id="pharyngeal.epithelia.2-looks-like-it-resembles-a-rather-unspecific-cluster-which-could-be-a-hint-for-bad-filtering.-cluster-doesnt-look-over--or-underclustered.">pharyngeal.epithelia.2 looks like it resembles a rather unspecific cluster, which could be a hint for bad filtering. Cluster doesn<c2><b4>t look over- or underclustered.</b4></c2></h5>
<p>Plot Top 5 DEGs and UMAP</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(</span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>    DEG.plot <span class="sc">&amp;</span> <span class="fu">DimPlot</span>(data1,</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">label =</span> T,</span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">reduction =</span> <span class="st">'umap'</span>,</span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">cols =</span> clust.cp.separate) <span class="sc">+</span></span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">NoAxes</span>() <span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="gene-expression-patterns" class="level4">
<h4 class="anchored" data-anchor-id="gene-expression-patterns">Gene expression patterns</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>  goi <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Six3-6'</span>,</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>          <span class="st">'FGFa2'</span>,</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>          <span class="st">'Wnt2'</span>,</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a>          <span class="st">'SnailA'</span>,</span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>          <span class="st">'Wnt3'</span>,</span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a>          <span class="st">'ERG'</span>,</span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a>          <span class="st">'Brachyury'</span>,</span>
<span id="cb119-8"><a href="#cb119-8" aria-hidden="true" tabindex="-1"></a>          <span class="st">'FoxA'</span>,</span>
<span id="cb119-9"><a href="#cb119-9" aria-hidden="true" tabindex="-1"></a>          <span class="st">'PCNA'</span>)</span>
<span id="cb119-10"><a href="#cb119-10" aria-hidden="true" tabindex="-1"></a>  <span class="co">#here you list the genes you want to plot; I would not plot more than 9-12 at a time...</span></span>
<span id="cb119-11"><a href="#cb119-11" aria-hidden="true" tabindex="-1"></a>  <span class="co">#* </span><span class="al">NOTE</span><span class="co">: these **MUST** be the same names as in your dataset!!</span></span>
<span id="cb119-12"><a href="#cb119-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeaturePlot</span>(</span>
<span id="cb119-13"><a href="#cb119-13" aria-hidden="true" tabindex="-1"></a>    data1,</span>
<span id="cb119-14"><a href="#cb119-14" aria-hidden="true" tabindex="-1"></a>    goi,</span>
<span id="cb119-15"><a href="#cb119-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">order =</span> T,</span>
<span id="cb119-16"><a href="#cb119-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cols = gene.cp, #gradients aren't working for me in Vs.5</span></span>
<span id="cb119-17"><a href="#cb119-17" aria-hidden="true" tabindex="-1"></a>    <span class="co"># reduction = 'umap.liger' #to use with other reductions</span></span>
<span id="cb119-18"><a href="#cb119-18" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">&amp;</span></span>
<span id="cb119-19"><a href="#cb119-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># remove the axis and legend; can omit these if you want to show them</span></span>
<span id="cb119-20"><a href="#cb119-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">NoAxes</span>() <span class="sc">&amp;</span> <span class="fu">NoLegend</span>() <span class="sc">&amp;</span><span class="fu">scale_colour_gradientn</span>(<span class="at">colours =</span>gene.cp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="naming-the-clusters" class="level4">
<h4 class="anchored" data-anchor-id="naming-the-clusters">Naming the clusters</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a> clusterNames<span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"celltypemarkers.xlsx"</span>)</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>clusterNames<span class="sc">$</span>Marker[<span class="dv">23</span>]<span class="ot">=</span><span class="st">'Ncol3'</span></span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>  goi <span class="ot">=</span> clusterNames<span class="sc">$</span>Marker</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#if cluster names are not integers: you tried this already and want to update / change it</span></span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>  data1 <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(data1, <span class="at">value =</span> <span class="st">'seurat_clusters'</span>)</span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a>  data1 <span class="ot">&lt;-</span></span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">BuildClusterTree</span>(data1, <span class="at">reorder =</span> T, <span class="at">reorder.numeric =</span> T) <span class="co">#this re-sets everything to numeric; required by script below</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The `slot` argument of `AverageExpression()` is deprecated as of Seurat 5.0.0.
i Please use the `layer` argument instead.
i The deprecated feature was likely used in the Seurat package.
  Please report the issue at &lt;https://github.com/satijalab/seurat/issues&gt;.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>As of Seurat v5, we recommend using AggregateExpression to perform pseudo-bulk analysis.
First group.by variable `ident` starts with a number, appending `g` to ensure valid variable names
Reordering identity classes and rebuilding tree

This message is displayed once per session.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DotPlot</span>(data1, <span class="at">features =</span> goi) <span class="sc">+</span> <span class="fu">RotatedAxis</span>() <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="co">#checks that all your genes are available and actually express specifically; useful for building the gene list</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>  <span class="co">#plot your data so you can see which cluster was which:</span></span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DimPlot</span>(data1, <span class="at">label =</span> T, <span class="at">cols =</span> clust.cp.separate) <span class="sc">+</span> <span class="fu">NoAxes</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-33-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a>  <span class="co">#assign cluster ID to the individual libraries:</span></span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a>  data1 <span class="ot">&lt;-</span></span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ScaleData</span>(data1, <span class="at">features =</span> goi, <span class="at">split.by =</span> <span class="st">'orig.ident'</span>) <span class="co">#works with scaled gene values, so must scale all of these first (This is just in case they are not within the VarFeatures set!: if you select candidates from DEG lists, then this is unnecessary)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Centering and scaling data matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Different features in new layer data than already exists for
scale.data</code></pre>
</div>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>  <span class="co">#set empty name variable:</span></span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>  clName <span class="ot">=</span> <span class="fu">vector</span>()</span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#generate a matrix of values of each cluster for each gene:</span></span>
<span id="cb129-5"><a href="#cb129-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Seurat has a function for this</span></span>
<span id="cb129-6"><a href="#cb129-6" aria-hidden="true" tabindex="-1"></a>  m <span class="ot">=</span> <span class="fu">AverageExpression</span>(data1, <span class="at">features =</span> goi, <span class="at">layer =</span> <span class="st">'scale.data'</span>)</span>
<span id="cb129-7"><a href="#cb129-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb129-8"><a href="#cb129-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># re-arranges the gene list in vs5: put it back in original order </span></span>
<span id="cb129-9"><a href="#cb129-9" aria-hidden="true" tabindex="-1"></a>  gind<span class="ot">=</span><span class="fu">match</span>(goi,<span class="fu">rownames</span>(m<span class="sc">$</span>RNA))</span>
<span id="cb129-10"><a href="#cb129-10" aria-hidden="true" tabindex="-1"></a>  m<span class="sc">$</span>RNA<span class="ot">=</span>m<span class="sc">$</span>RNA[gind,]</span>
<span id="cb129-11"><a href="#cb129-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb129-12"><a href="#cb129-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">levels</span>(data1<span class="sc">@</span>active.ident)))</span>
<span id="cb129-13"><a href="#cb129-13" aria-hidden="true" tabindex="-1"></a>    <span class="co">#for each cluster set</span></span>
<span id="cb129-14"><a href="#cb129-14" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb129-15"><a href="#cb129-15" aria-hidden="true" tabindex="-1"></a>    clName[j] <span class="ot">=</span> <span class="fu">as.integer</span>(<span class="fu">which.max</span>(m<span class="sc">$</span>RNA[, j])) <span class="co">#choose the highest value</span></span>
<span id="cb129-16"><a href="#cb129-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb129-17"><a href="#cb129-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sort</span>(clName) <span class="co">#this prints which IDs were selected; sanity check that it did what you wanted... should have no unwanted replicated IDs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1]  1  2  6  8 10 12 15 20 24 27</code></pre>
</div>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>  clusterNames<span class="sc">$</span>ID[clName] <span class="co">#you can also look at the names to see where the problem might have been</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "cnidocyte.2"            "digestive.gland.2"      "mucin.gland"           
 [4] "neuron.2"               "mature.cnidocyte"       "gastrodermis.2"        
 [7] "embryonic.ectoderm.2"   "embryonic.ectoderm"     "epithelia.2"           
[10] "pharyngeal.epithelia.2"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>  <span class="co">#first order the identities..</span></span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>  data1<span class="sc">@</span>active.ident <span class="ot">=</span> <span class="fu">factor</span>(data1<span class="sc">@</span>active.ident,</span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">levels</span>(data1<span class="sc">@</span>active.ident)[<span class="fu">order</span>(clName)])</span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#set the names to your IDs; once the gene list works, it is really easy to then update your cluster names with your excel spreadsheet.</span></span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">levels</span>(data1<span class="sc">@</span>active.ident) <span class="ot">=</span> clusterNames<span class="sc">$</span>ID[clName][<span class="fu">order</span>(clName)]</span>
<span id="cb133-6"><a href="#cb133-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb133-7"><a href="#cb133-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#save the IDs in metadata:</span></span>
<span id="cb133-8"><a href="#cb133-8" aria-hidden="true" tabindex="-1"></a>  data1<span class="sc">@</span>meta.data<span class="sc">$</span>IDs <span class="ot">=</span> data1<span class="sc">@</span>active.ident</span>
<span id="cb133-9"><a href="#cb133-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb133-10"><a href="#cb133-10" aria-hidden="true" tabindex="-1"></a>  <span class="co">#plot to confirm all is as expected:</span></span>
<span id="cb133-11"><a href="#cb133-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DimPlot</span>(data1,</span>
<span id="cb133-12"><a href="#cb133-12" aria-hidden="true" tabindex="-1"></a>          <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">'IDs'</span>),</span>
<span id="cb133-13"><a href="#cb133-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">label =</span> T,</span>
<span id="cb133-14"><a href="#cb133-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">cols =</span> clust.cp.separate) <span class="sc">+</span> <span class="fu">NoAxes</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-33-3.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="a-more-permissive-differential-gene-expression-as-a-preparation-for-goterm" class="level4">
<h4 class="anchored" data-anchor-id="a-more-permissive-differential-gene-expression-as-a-preparation-for-goterm">A more permissive differential gene expression as a preparation for GoTerm</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a>data1<span class="sc">@</span>active.assay <span class="ot">=</span> <span class="st">'RNA'</span></span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a>    all.markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a>      data1,</span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">logfc.threshold =</span> <span class="fl">0.1</span>,</span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">features =</span> <span class="cn">NULL</span>,</span>
<span id="cb134-6"><a href="#cb134-6" aria-hidden="true" tabindex="-1"></a>      <span class="co">#this is faster but restricted to variable gene set.</span></span>
<span id="cb134-7"><a href="#cb134-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">only.pos =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster embryonic.ectoderm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster embryonic.ectoderm.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster epithelia.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster gastrodermis.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster pharyngeal.epithelia.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster digestive.gland.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster mucin.gland</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster neuron.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster cnidocyte.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster mature.cnidocyte</code></pre>
</div>
</div>
</section>
<section id="running-goterm" class="level4">
<h4 class="anchored" data-anchor-id="running-goterm">Running GoTerm</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"topGO"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a>  {BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"topGO"</span>,<span class="at">lib=</span><span class="st">"/lisc/user/tillmann"</span>)}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SeuratObject':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,
    as.data.frame, basename, cbind, colnames, dirname, do.call,
    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,
    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,
    tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'graph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:ape':

    complement, degree, edges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    I, expand.grid, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:sp':

    %over%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationDbi'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SparseM'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    backsolve</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
groupGOTerms:   GOBPTerm, GOMFTerm, GOCCTerm environments built.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'topGO'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    members</code></pre>
</div>
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(topGO)</span>
<span id="cb169-2"><a href="#cb169-2" aria-hidden="true" tabindex="-1"></a>  {    </span>
<span id="cb169-3"><a href="#cb169-3" aria-hidden="true" tabindex="-1"></a>    Unfold <span class="ot">&lt;-</span> annotations <span class="sc">%&gt;%</span> </span>
<span id="cb169-4"><a href="#cb169-4" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="st">`</span><span class="at">GO</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">strsplit</span>(<span class="fu">as.character</span>(<span class="st">`</span><span class="at">GO</span><span class="st">`</span>), <span class="st">","</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb169-5"><a href="#cb169-5" aria-hidden="true" tabindex="-1"></a>      tidyr<span class="sc">::</span><span class="fu">unnest</span>(<span class="st">`</span><span class="at">GO</span><span class="st">`</span>) </span>
<span id="cb169-6"><a href="#cb169-6" aria-hidden="true" tabindex="-1"></a>    geneID2GO <span class="ot">&lt;-</span> Unfold <span class="sc">%&gt;%</span> <span class="fu">split</span>(<span class="at">x =</span> .<span class="sc">$</span><span class="st">`</span><span class="at">GO</span><span class="st">`</span>, <span class="at">f =</span> .<span class="sc">$</span>gene_short_name)</span>
<span id="cb169-7"><a href="#cb169-7" aria-hidden="true" tabindex="-1"></a>    geneNames <span class="ot">&lt;-</span> <span class="fu">names</span>(geneID2GO)</span>
<span id="cb169-8"><a href="#cb169-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-9"><a href="#cb169-9" aria-hidden="true" tabindex="-1"></a>    <span class="co">#process data</span></span>
<span id="cb169-10"><a href="#cb169-10" aria-hidden="true" tabindex="-1"></a>    GOdata <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb169-11"><a href="#cb169-11" aria-hidden="true" tabindex="-1"></a>    myInterestingGenes <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb169-12"><a href="#cb169-12" aria-hidden="true" tabindex="-1"></a>    geneList <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb169-13"><a href="#cb169-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this takes a really long time: could be done via SLURM</span></span>
<span id="cb169-14"><a href="#cb169-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">unique</span>(all.markers<span class="sc">$</span>cluster)))</span>
<span id="cb169-15"><a href="#cb169-15" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb169-16"><a href="#cb169-16" aria-hidden="true" tabindex="-1"></a>      <span class="co">#identify the GOI from the DE list</span></span>
<span id="cb169-17"><a href="#cb169-17" aria-hidden="true" tabindex="-1"></a>      myInterestingGenes[[i]] <span class="ot">&lt;-</span> all.markers<span class="sc">$</span>gene[all.markers<span class="sc">$</span>cluster<span class="sc">==</span><span class="fu">unique</span>(all.markers<span class="sc">$</span>cluster)[i]] <span class="co">#list of genes you want to perform GO enrichment for</span></span>
<span id="cb169-18"><a href="#cb169-18" aria-hidden="true" tabindex="-1"></a>      geneList[[i]] <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.integer</span>(geneNames <span class="sc">%in%</span> myInterestingGenes[[i]]))</span>
<span id="cb169-19"><a href="#cb169-19" aria-hidden="true" tabindex="-1"></a>      <span class="fu">names</span>(geneList[[i]]) <span class="ot">&lt;-</span> geneNames</span>
<span id="cb169-20"><a href="#cb169-20" aria-hidden="true" tabindex="-1"></a>      GOdata[[i]] <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">"topGOdata"</span>,<span class="at">ontology =</span> <span class="st">"BP"</span>, <span class="at">allGenes =</span> geneList[[i]],<span class="at">annot =</span> annFUN.gene2GO, <span class="at">gene2GO =</span> geneID2GO)</span>
<span id="cb169-21"><a href="#cb169-21" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb169-22"><a href="#cb169-22" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb169-23"><a href="#cb169-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter and generate figures.</span></span>
<span id="cb169-24"><a href="#cb169-24" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb169-25"><a href="#cb169-25" aria-hidden="true" tabindex="-1"></a>    resultFis<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb169-26"><a href="#cb169-26" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">unique</span>(all.markers<span class="sc">$</span>cluster)))</span>
<span id="cb169-27"><a href="#cb169-27" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb169-28"><a href="#cb169-28" aria-hidden="true" tabindex="-1"></a>      <span class="co">#run the test...</span></span>
<span id="cb169-29"><a href="#cb169-29" aria-hidden="true" tabindex="-1"></a>      resultFis[[i]] <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GOdata[[i]], <span class="at">algorithm =</span> <span class="st">"classic"</span>, <span class="at">statistic =</span> <span class="st">"fisher"</span>) </span>
<span id="cb169-30"><a href="#cb169-30" aria-hidden="true" tabindex="-1"></a>      pvalFis <span class="ot">&lt;-</span> <span class="fu">score</span>(resultFis[[i]])</span>
<span id="cb169-31"><a href="#cb169-31" aria-hidden="true" tabindex="-1"></a>      <span class="co">#filter for only &gt;0.05</span></span>
<span id="cb169-32"><a href="#cb169-32" aria-hidden="true" tabindex="-1"></a>      pvalFis <span class="ot">=</span> pvalFis[pvalFis<span class="sc">&gt;=</span><span class="fl">0.05</span>]</span>
<span id="cb169-33"><a href="#cb169-33" aria-hidden="true" tabindex="-1"></a>      allRes_intgenes<span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GOdata[[i]], <span class="at">pvalues =</span> resultFis[[i]], <span class="at">orderBy =</span> <span class="st">"pvalues"</span>, <span class="at">topNodes=</span><span class="dv">30</span>)</span>
<span id="cb169-34"><a href="#cb169-34" aria-hidden="true" tabindex="-1"></a>      allRes_intgenes<span class="sc">$</span>pvalues<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(allRes_intgenes<span class="sc">$</span>pvalues)</span>
<span id="cb169-35"><a href="#cb169-35" aria-hidden="true" tabindex="-1"></a>      <span class="co">#convert NA values to zero; only if p value is so small it cannot be displayed in r</span></span>
<span id="cb169-36"><a href="#cb169-36" aria-hidden="true" tabindex="-1"></a>      allRes_intgenes[<span class="fu">is.na</span>(allRes_intgenes)]<span class="ot">&lt;-</span><span class="fl">0.00000001</span></span>
<span id="cb169-37"><a href="#cb169-37" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb169-38"><a href="#cb169-38" aria-hidden="true" tabindex="-1"></a>      <span class="co">#plot GOenrichment </span></span>
<span id="cb169-39"><a href="#cb169-39" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb169-40"><a href="#cb169-40" aria-hidden="true" tabindex="-1"></a>      colCP<span class="ot">=</span><span class="fu">c</span>(<span class="st">"darkred"</span>)</span>
<span id="cb169-41"><a href="#cb169-41" aria-hidden="true" tabindex="-1"></a>      p[[i]]<span class="ot">=</span></span>
<span id="cb169-42"><a href="#cb169-42" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(allRes_intgenes, ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x=</span>(<span class="fu">reorder</span>(Term,(<span class="sc">-</span><span class="fu">log10</span>(pvalues)))), <span class="at">y=</span>(<span class="sc">-</span><span class="fu">log10</span>(pvalues)))) <span class="sc">+</span></span>
<span id="cb169-43"><a href="#cb169-43" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">stat_summary</span>(<span class="at">geom =</span> <span class="st">"bar"</span>, <span class="at">fun =</span> mean, <span class="at">position =</span> <span class="st">"dodge"</span>,</span>
<span id="cb169-44"><a href="#cb169-44" aria-hidden="true" tabindex="-1"></a>                              <span class="at">col=</span>colCP,<span class="at">fill=</span>colCP) <span class="sc">+</span></span>
<span id="cb169-45"><a href="#cb169-45" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb169-46"><a href="#cb169-46" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">xlab</span>(<span class="st">"Biological Process"</span>) <span class="sc">+</span></span>
<span id="cb169-47"><a href="#cb169-47" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">ylab</span>(<span class="st">"Enrichment -log10 p-value"</span>) <span class="sc">+</span></span>
<span id="cb169-48"><a href="#cb169-48" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">title=</span><span class="st">"GO enrichment"</span>,<span class="at">subtitle=</span><span class="fu">levels</span>(all.markers<span class="sc">$</span>cluster)[i])</span>
<span id="cb169-49"><a href="#cb169-49" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb169-50"><a href="#cb169-50" aria-hidden="true" tabindex="-1"></a>  } </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 10705 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 12441 GO terms and 27196 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13894 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 10705 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 12441 GO terms and 27196 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13894 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 10705 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 12441 GO terms and 27196 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13894 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 10705 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 12441 GO terms and 27196 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13894 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 10705 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 12441 GO terms and 27196 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13894 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 10705 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 12441 GO terms and 27196 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13894 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 10705 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 12441 GO terms and 27196 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13894 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 10705 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 12441 GO terms and 27196 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13894 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 10705 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 12441 GO terms and 27196 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13894 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 10705 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 12441 GO terms and 27196 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13894 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 7926 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: NAs introduced by coercion</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 5241 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: NAs introduced by coercion</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 8466 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 8351 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 8994 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: NAs introduced by coercion</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 6920 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 6928 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 8239 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: NAs introduced by coercion</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 6775 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 8689 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="sourceCode cell-code" id="cb244"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb244-1"><a href="#cb244-1" aria-hidden="true" tabindex="-1"></a>  p[[<span class="dv">1</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb245"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb245-1"><a href="#cb245-1" aria-hidden="true" tabindex="-1"></a>  p[[<span class="dv">2</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-35-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb246"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb246-1"><a href="#cb246-1" aria-hidden="true" tabindex="-1"></a>  p[[<span class="dv">3</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-35-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb247"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb247-1"><a href="#cb247-1" aria-hidden="true" tabindex="-1"></a>  p[[<span class="dv">4</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-35-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb248"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb248-1"><a href="#cb248-1" aria-hidden="true" tabindex="-1"></a>  p[[<span class="dv">5</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-35-5.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb249"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb249-1"><a href="#cb249-1" aria-hidden="true" tabindex="-1"></a>  p[[<span class="dv">6</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-35-6.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb250"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb250-1"><a href="#cb250-1" aria-hidden="true" tabindex="-1"></a>  p[[<span class="dv">7</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-35-7.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb251"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb251-1"><a href="#cb251-1" aria-hidden="true" tabindex="-1"></a>  p[[<span class="dv">8</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-35-8.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb252"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb252-1"><a href="#cb252-1" aria-hidden="true" tabindex="-1"></a>  p[[<span class="dv">9</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-35-9.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a>  p[[<span class="dv">10</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-35-10.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb254"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb254-1"><a href="#cb254-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="go-enrichment---embryonic-ectoderm" class="level4">
<h4 class="anchored" data-anchor-id="go-enrichment---embryonic-ectoderm">Go Enrichment - embryonic ectoderm</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb255"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb255-1"><a href="#cb255-1" aria-hidden="true" tabindex="-1"></a>p[[<span class="dv">1</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The Embryonic Ectoderm is characterized by a lot of cillium production, that could mean that the cells of this cluster represent a part of the dermis, that has cillia.</p>
</section>
<section id="go-enrichment---embryonic-ectoderm.2" class="level4">
<h4 class="anchored" data-anchor-id="go-enrichment---embryonic-ectoderm.2">Go Enrichment - embryonic ectoderm.2</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb256"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb256-1"><a href="#cb256-1" aria-hidden="true" tabindex="-1"></a>p[[<span class="dv">2</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Embryonic ectoderm 2 is very similar to embryonic ectoderm, so maybe this could be a sign of overclustering.</p>
</section>
<section id="go-enrichment---epithelia.2" class="level4">
<h4 class="anchored" data-anchor-id="go-enrichment---epithelia.2">Go enrichment - epithelia.2</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb257"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb257-1"><a href="#cb257-1" aria-hidden="true" tabindex="-1"></a>p[[<span class="dv">3</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="go-enrichment---gastrodermis.2" class="level4">
<h4 class="anchored" data-anchor-id="go-enrichment---gastrodermis.2">Go enrichment - gastrodermis.2</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb258"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb258-1"><a href="#cb258-1" aria-hidden="true" tabindex="-1"></a>p[[<span class="dv">4</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Gastrodermis has a lot of actin and cytoskeletal processes because it is able to contract</p>
</section>
<section id="go---enrichment---pharyngeal.epithelia.2" class="level4">
<h4 class="anchored" data-anchor-id="go---enrichment---pharyngeal.epithelia.2">Go - Enrichment - pharyngeal.epithelia.2</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb259"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb259-1"><a href="#cb259-1" aria-hidden="true" tabindex="-1"></a>p[[<span class="dv">5</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="go---enrichment---digestive.gland.2" class="level4">
<h4 class="anchored" data-anchor-id="go---enrichment---digestive.gland.2">Go - Enrichment - digestive.gland.2</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb260"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb260-1"><a href="#cb260-1" aria-hidden="true" tabindex="-1"></a>p[[<span class="dv">6</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Digstive gland has a lot of peptide and protein building processes, which fits nicely in the scheme of gigstive glands.</p>
</section>
<section id="go---enrichment---mucin.gland" class="level4">
<h4 class="anchored" data-anchor-id="go---enrichment---mucin.gland">Go - Enrichment - mucin.gland</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb261"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb261-1"><a href="#cb261-1" aria-hidden="true" tabindex="-1"></a>p[[<span class="dv">7</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The mucin gland has a lt of glycoprotein and transport processes and fits nicely with their function of producing mucus and transporting it out of the cell</p>
</section>
<section id="go---enrichment---neuron.2" class="level4">
<h4 class="anchored" data-anchor-id="go---enrichment---neuron.2">Go - Enrichment - neuron.2</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb262"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb262-1"><a href="#cb262-1" aria-hidden="true" tabindex="-1"></a>p[[<span class="dv">8</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Neuron is also very conclusive, since we find processes linked with neuronal functiontions, like stimulus responses, behavior, signaling and so on.</p>
</section>
<section id="go---enrichment---cnidocyte.2" class="level4">
<h4 class="anchored" data-anchor-id="go---enrichment---cnidocyte.2">Go - Enrichment - cnidocyte.2</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb263"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb263-1"><a href="#cb263-1" aria-hidden="true" tabindex="-1"></a>p[[<span class="dv">9</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Cnidocytes are produced by fusing vesicles of the Golgi Apparatus, so the Golgi Apparatus processes fit very nicely into the function of the cnidocytes.</p>
</section>
<section id="go---enrichment---mature.cnydocyte" class="level4">
<h4 class="anchored" data-anchor-id="go---enrichment---mature.cnydocyte">Go - Enrichment - mature.cnydocyte</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb264"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb264-1"><a href="#cb264-1" aria-hidden="true" tabindex="-1"></a>p[[<span class="dv">10</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="quarto.tillmann_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Mature cnidocytes build long arrays of spines, which is mirrored here by the cilium and microtubule processes.</p>
</section>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>