# SCCourse2023W
Scripts and presentations of the single cell course <br>
1. STARTest:  aligns the reference genome of nematostella vectensiensis to the RNA Sequences SRR24348397<br>
2. trimmomatic: Trims aligned elemants in order to make the quality of the alignment better <br>
3. trim_star: in trim star we aligned only double stranded alignments to each other <br>
4. samtoolsfilter.sh: sam tools filter filters the reads <br>
5. stringtie: uses a genome-guided transcriptome assembly approach along with concepts from de novo genome assembly to improve transcript assembly <br>
6. augustus performs a ab initio gene prediction output<br> 
7. Rhistory.sh
8. cellranger.sh processes and analyzes single-cell RNA sequencing data.
9. emapper.sh: is a tool for fast functional annotation of novel sequences
10. featureCounts.sh: counts mapped reads for genomic features such as genes, exons, promoter, gene bodies, genomic bins and chromosomal locations
11. interproscan.sh: scans sequences against member database signatures
12. scripts160124.sh
